interp/notation_ops.ml

```diff
@@ -1344,7 +1361,7 @@ let rec match_ inner u alp metas sigma a1 a2 =
 
   (* Matching compositionally *)
   | GVar id1, NVar id2 when alpha_var id1 id2 (fst (snd alp)) -> sigma
-  | GRef (r1,_), NRef r2 when (GlobRef.equal r1 r2) -> sigma
+  | GRef (r1,u1), NRef (r2,u2) when (GlobRef.equal r1 r2) && compare_glob_universe_instances false u1 u2 -> sigma
```

Comment:
As long as notations don't have variables for denoting universes, I believe this should be an equality (possibly up to the order of variables).

Reply:
I think we do want the `<=` relation here. We want a notation `foo` to match a concrete term `foo@{i j}`, and likewise notation `foo@{Top.1 _}` to match `foo@{Top.1 Top.2}`.

Reply:
Forget my comment.  You're right. Order matters, and `_` should be less than `i` and nothing less than something.